src/fetcher.js

```diff
@@ -65,7 +65,7 @@ export default class Fetcher {
         this.authenticationErrorHandler();
       }
 
-      throw new Error(errorMessage);
+      throw new Error(JSON.stringify({'type': errorMessage, 'message': await resp.text()}));
```

Comment:
Instead of cramming a data structure into a json string, it might be better to extend Error with our own error type with the additional data.